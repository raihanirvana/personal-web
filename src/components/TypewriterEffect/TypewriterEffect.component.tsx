import React from 'react';

import GeneralText from '../GeneralText';

const useTypingEffect = (
  index: number, setIndex: React.Dispatch<React.SetStateAction<number>>, 
  text: string, setTypedText: React.Dispatch<React.SetStateAction<string>>
) => {
  React.useEffect(() => {
    if (index < text.length) {
      const timeout = setTimeout(() => {
        setTypedText((prev) => prev + text[index]);
        setIndex((prev) => prev + 1);
      }, 300);

      return () => clearTimeout(timeout);
    }
  }, [index, text]);
};

const useTypewriterEffect = (text: string) => {
  const [typedText, setTypedText] = React.useState("");
  const [index, setIndex] = React.useState(0);

  useTypingEffect(index, setIndex, text, setTypedText);

  return typedText;
};

const TypewriterEffect = ({ text, customStyle }: { text: string; customStyle?: string }) => {
  const typedText = useTypewriterEffect(text);

  return <GeneralText text={typedText} customStyle={customStyle} />;
};

export default TypewriterEffect;
