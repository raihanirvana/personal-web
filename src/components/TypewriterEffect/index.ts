export { default } from './TypewriterEffect.component'