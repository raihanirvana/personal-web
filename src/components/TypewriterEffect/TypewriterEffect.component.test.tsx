import { render } from '@testing-library/react';
import TypewriterEffect from './TypewriterEffect.component.tsx'

describe('ContactsList Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<TypewriterEffect text='Raihan' customStyle={'text-white'} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
