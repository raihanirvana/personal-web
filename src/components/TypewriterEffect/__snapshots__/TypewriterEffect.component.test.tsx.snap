// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsList Component matches the snapshot 1`] = `
<DocumentFragment>
  <p
    class="text-white text-white"
  />
</DocumentFragment>
`;
